### ISSUE TITLE
Pkg.generate fails if `origin` remote already exists

### ISSUE THREAD

**Issue by huonw** [JuliaLang/julia#13527]
In my `~/.gitconfig` I tweak the branches that git tries to fetch for any remote called `origin` so I get all pull requests as well as normal branches:

```ini
[remote "origin"]
        fetch = +refs/pull/*/head:refs/remotes/origin/pr/*
        fetch = +refs/heads/*:refs/remotes/origin/*
```

This has the unfortunate side-effect of implicitly defining the `origin` remote in every repository, even newly created ones. `Pkg.generate` creates a new git repo and runs `git remote add origin ...`, which fails since `origin` "exists" already, causing `Pkg.generate` itself to fail too; it could possibly detect this case and run `git remote set-url origin ...` instead of `add`. (On the other hand, handling this slightly peculiar set-up of mine could easily be deemed too specialised. :) )

Output:

```
julia> Pkg.generate("Test", "MIT")
INFO: Initializing Test repo: /home/huon/.julia/v0.4/Test
INFO: Origin: git://github.com/huonw/Test.jl.git
fatal: remote origin already exists.
ERROR: failed process: Process(`git --work-tree=/home/huon/.julia/v0.4/Test --git-dir=/home/huon/.julia/v0.4/Test/.git remote add origin git://github.com/huonw/Test.jl.git`, ProcessExited(128)) [128]
 in pipeline_error at process.jl:555
 in run at process.jl:531
 in init at pkg/generate.jl:103
 in package at pkg/generate.jl:45
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
```

Version:

```
julia> versioninfo()
Julia Version 0.4.0
Commit 0ff703b* (2015-10-08 06:20 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i7-4900MQ CPU @ 2.80GHz
  WORD_SIZE: 64
  BLAS: libopenblas (NO_LAPACKE DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: liblapack.so.3
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

**Comment by tkelman** [comment]
Hm, we may need to make some of https://github.com/JuliaLang/julia/blob/1bb7e10ed26e4123066191c49079ea70ebb7ec1b/base/pkg/generate.jl#L103-L106 optional, or put a try-catch with a warning around them?

**Comment by jakebolewski** [comment]
Closing and moving to PkgDev.jl

**Comment by tkelman** [comment]
fix should ideally also be backported for 0.4 though
